src/Common/src/TypeSystem/Common/TargetDetails.cs

```diff
@@ -204,7 +206,6 @@ public LayoutInt GetObjectAlignment(LayoutInt fieldAlignment)
             {
                 case TargetArchitecture.ARM:
                 case TargetArchitecture.ARMEL:
-                case TargetArchitecture.Wasm32:
```

Comment:
@morganbr Can you please have a look at this? I'm not sure why wasm required `align8` for this. We either need to update the comment to say something about wasm too, or stop doing this (and take my change).

Reply:
I doubt it needed that. The semi-requirement on asm is that all accesses are aligned (wasm can technically do unaligned, but it's a lot slower. asm.js can't always do unaligned correctly). Please just test by doing a wasm build and ensuring the test passes and prints a bunch of stuff (requires #5492, otherwise you have to run manually in a browser).